src/public/modules/forms/helpers/CsvGenerator.ts

```diff
@@ -38,25 +47,25 @@ module.exports = class CsvGenerator {
    * Insert headers into the CSV file after the meta-data
    * @param {Array} headerLabels array of labels for header row
    */
-  setHeader(headerLabels) {
+  setHeader(headerLabels: (number | string)[]): void {
     this.records[this.startIdx - 1] = stringify(headerLabels)
   }
 
   /**
    * Insert meta-data array into the start of the CSV file
    * @param {Array} metaDataRows array of arrays, metaDataRows[i][j] holds the data for row i, col j of the metaData table
    */
-  addMetaData(metaDataRows) {
+  addMetaData(metaDataRows: (number | string)[]): void {
```

Comment:
I'm not sure this typing is correct. the JSdocs says "array of arrays", and I can see e.g. from `CsvMergedHeadersGenerator.js` that the argument being passed is of type `(number | string)[][]`

Reply:
edited